### PR TITLE
ignition: propagate proxy settings from a host into a vm

### LINKED ISSUE #13168: Leak the HTTP*PROXY environment variables into the VM when doing a podman machine start.
Leak the following environment variables
```
var ProxyEnv = []string{
	"http_proxy",
	"https_proxy",
	"ftp_proxy",
	"no_proxy",
	"HTTP_PROXY",
	"HTTPS_PROXY",
	"FTP_PROXY",
	"NO_PROXY",
}
```
Into the VM at start time.

<@cyberpear> systemd will set them if you pass on the kernel CMDLINE `systemd.setenv=VARIABLE=VALUE` at boot

### PR DESCRIPTION
**Pull request by esendjer**
PR solves the need of setting proxy for systemd and the whole guest system.

Set proxy settings (such as `HTTP_PROXY`, and others) for the whole guest OS with setting up `DefaultEnvironment` with a `systemd` configuration file `default-env.conf`, a `profile.d` scenario file - `default-env.sh` and a `environment.d` configuration file `default-env.conf`.

The **actual** environment variables are read by podman at a start, then they are encrypted with base64 into a single string and after are provided into a VM through QEMU Firmware Configuration (fw_cfg) Device

Inside a VM a systemd service `envset-fwcfg.service` reads the providead encrypted string from fw_cfg, decrypts and then adds to the files:
 - `/etc/systemd/system.conf.d/default-env.conf`
 - `/etc/profile.d/default-env.sh`
 - `/etc/environment.d/default-env.conf`

At the end, this service execute  `systemctl daemon-reload` to propagate new variables for systemd manager

[NO NEW TESTS NEEDED]

Closes #13168

Also related to #11941 #12739

### CODE REVIEW

File: pkg/machine/ignition.go [hunk 1]
```diff
@@ -506,14 +534,14 @@ func prepareCertFile(path string, name string) (File, error) {
 	return file, nil
 }
 
-func getProxyVariables() string {
-	proxyOpts := ""
+func GetProxyVariables() *map[string]string {
```
Comment (mheon) [inline pkg/machine/ignition.go:537]: Why pointer to a map? Maps are already reference types

Comment (esendjer) [reply pkg/machine/ignition.go]: Good notice, thanks. Fixed

**Comment by baude** [comment]
LGTM

**Comment by ctml91** [comment]
Any trick to making this work on a mac?

```
podman version 4.0.2
```

```
$ podman machine stop
$ podman machine rm
$ env | grep -i proxy
HTTPS_PROXY=myproxy
HTTP_PROXY=myproxy
https_proxy=myproxy
http_proxy=myproxy
$ podman machine init
$ podman machine list
NAME                     VM TYPE     CREATED         LAST UP            CPUS        MEMORY      DISK SIZE
podman-machine-default*  qemu        2 minutes ago  Currently running  1           2.147GB     107.4GB
$ podman build -t test:latest ./
Error: error creating build container: initializing source docker://registry.access.redhat.com/ubi8/ubi-minimal:latest: pinging container registry registry.access.redhat.com: Get "https://registry.access.redhat.com/v2/": x509: certificate signed by unknown authority
```

This happens when requests do not go to proxy and go to our honeypot instead.

**Comment by amilanoski** [comment]
This is still an issue with MacOS Ventura 13.3.1 with fresh installation and Corporate proxies set in my zsh shell.

The only thing that worked was having to manually  run
1. podman machine ssh then run the configuration setup found here. https://github.com/containers/podman/issues/11941#issuecomment-997279722

I have not found any other documentation around that can solve this with out having users manually go in and configure this.  Keeping the scope only with proxies these should be pulled into the **podman machine init** or better yet **podman machine start**.

**Comment by esendjer** [comment]
@amilanoski 
I see. Let me check it on my Macs.
But until I've done, could you please share how do you set proxies in your zsh shell. Here, I'm wondering commands and variable names, it allows me to repeat this on my side.

**Comment by esendjer** [comment]
The check is done, and unfortunately the problem has come back.
A brief research led me to the next:
* the issue was not until v4.5.1 release
* as far as I can tell, the issue is caused by some of the changes within the PR #18167

I want to dive deeper to understand how it could be solved.

**Comment by esendjer** [comment]
@amilanoski FYI
PR that fixes the issue #18759 has just been opened

**Pull request merged**